### DISCUSSION TITLE
在调公用方法时如何传组件的属性值

### DISCUSSION THREAD

**Discussion by stwflyfox** [opentiny/tiny-engine#889] [Q&A]
在调公用方法时如何传组件的属性值？
![image](https://github.com/user-attachments/assets/e74ea065-a5c2-48a3-8abd-11327dac1bcd)

![image](https://github.com/user-attachments/assets/2df7c692-47d0-4b17-97c5-0ce59b783b15)

**Comment by chilingling** [comment]
1. 如果组件的属性值绑定了变量，那么直接传递该绑定的变量即可，比如 `this.utils.setAuthorization(this.state.gm_category)`
2. 如果是循环渲染中的一些变量，那么直接就可以拿到，比如 `this.utils.SetAuthorization(item.xxx)`

**Reply by stwflyfox** [reply]
这个组件的属性值， 有可能不绑定变量直接输入的。这样的情况能取到吗？

**Reply by chilingling** [reply]
不绑定变量的情况，我理解无法直接拿到对应的值。
比如，在高代码中，我们有一个组件：

```vue
<template>
	<tiny-button v-if="这里读取 type 属性进行渲染" type="primary">
          <icon v-if="或者这里读取 type 属性进行渲染">
    </tiny-button>
</template>
```

这里，如果是直接的一些属性赋值，不通过变量，我们一般情况下是没法拿到的。
当然，还有一种方法：那就是通过  ref 属性，拿到组件的实例。
比如 `tiny-grid` 表格组件，就可以通过 `ref` 属性，拿到表格暴露出来的方法，属性等。
具体可以参考一下 tiny-vue 表格组件示例。
或者是 vue 文档（[expose 属性](https://vuejs.org/api/options-state.html#expose)）

**Comment by stwflyfox** [comment]
thanks
